File: android/employer/app/src/main/java/com/workforcex/employer/api/WorkforceXApi.java
package com.workforcex.employer.api;

import java.util.List;
import retrofit2.Call;
import retrofit2.http.*;

public interface WorkforceXApi {

    @POST("api/auth/register")
    Call<RegisterResponse> register(@Body RegisterRequest request);

    @POST("api/auth/login")
    Call<LoginResponse> login(@Body LoginRequest request);

    @PUT("api/employer/profile")
    Call<EmployerProfileResponse> saveEmployerProfile(
            @Header("Authorization") String token, @Body EmployerProfileRequest request);

    @GET("api/employer/profile")
    Call<EmployerProfileResponse> getEmployerProfile(@Header("Authorization") String token);

    @POST("api/jobs")
    Call<JobResponse> createJob(@Header("Authorization") String token, @Body JobRequest request);

    @GET("api/jobs")
    Call<List<JobResponse>> getMyJobs(@Header("Authorization") String token);

    @GET("api/jobs/{jobId}")
    Call<JobResponse> getJobById(@Header("Authorization") String token, @Path("jobId") String jobId);

    @PUT("api/jobs/{jobId}")
    Call<JobResponse> updateJob(@Header("Authorization") String token,
            @Path("jobId") String jobId, @Body JobRequest request);

    @DELETE("api/jobs/{jobId}")
    Call<Void> deleteJob(@Header("Authorization") String token, @Path("jobId") String jobId);

    @GET("api/matching/{jobId}")
    Call<List<MatchedWorker>> getMatchedWorkers(@Header("Authorization") String token,
            @Path("jobId") String jobId);

    @POST("api/matching/search") // Changed to POST
    Call<List<CandidateSearchResult>> searchCandidates(
            @Header("Authorization") String token,
            @Body CandidateSearchRequest request); // Changed to @Body

    // Applications
    @GET("api/applications/job/{jobId}")
    Call<List<JobApplicationItem>> getApplicationsForJob(
            @Header("Authorization") String token, @Path("jobId") String jobId);

    @PUT("api/applications/{applicationId}/status")
    Call<JobApplicationItem> updateApplicationStatus(
            @Header("Authorization") String token,
            @Path("applicationId") String applicationId,
            @Query("status") String status);

    @POST("api/applications/offer")
    Call<JobApplicationItem> offerJob(
            @Header("Authorization") String token,
            @Query("jobId") String jobId,
            @Query("workerId") String workerId);

    // Notifications
    @GET("api/notifications")
    Call<List<Notification>> getUnreadNotifications(@Header("Authorization") String token);

    @PUT("api/notifications/{notificationId}/read")
    Call<Void> markAsRead(@Header("Authorization") String token, @Path("notificationId") String notificationId);

    // Verification
    @GET("api/verification/pending")
    Call<List<Verification>> getPendingVerifications(@Header("Authorization") String token);

    @GET("api/verification/worker/{workerId}")
    Call<List<Verification>> getWorkerDocuments(@Header("Authorization") String token, @Path("workerId") String workerId);

    @PUT("api/verification/{verificationId}/status")
    Call<Void> updateEmployerVerificationStatus(
            @Header("Authorization") String token,
            @Path("verificationId") String verificationId,
            @Query("status") String status,
            @Body String comments);

    // Skills
    @GET("api/skills")
    Call<List<Skill>> getSkills();
}

File: android/employer/shared_employer/src/commonMain/kotlin/com/workforcex/shared_employer/WorkforceXApi.kt
package com.workforcex.shared_employer

import com.workforcex.shared_employer.models.*
import retrofit2.Call
import retrofit2.http.*

interface WorkforceXApi {

    @POST("api/auth/register")
    fun register(@Body request: RegisterRequest): Call<RegisterResponse>

    @POST("api/auth/login")
    fun login(@Body request: LoginRequest): Call<LoginResponse>

    @PUT("api/employer/profile")
    fun saveEmployerProfile(
        @Header("Authorization") token: String, @Body request: EmployerProfileRequest
    ): Call<EmployerProfileResponse>

    @GET("api/employer/profile")
    fun getEmployerProfile(@Header("Authorization") token: String): Call<EmployerProfileResponse>

    @POST("api/jobs")
    fun createJob(@Header("Authorization") token: String, @Body request: JobRequest): Call<JobResponse>

    @GET("api/jobs")
    fun getMyJobs(@Header("Authorization") token: String): Call<List<JobResponse>>

    @GET("api/jobs/{jobId}")
    fun getJobById(@Header("Authorization") token: String, @Path("jobId") jobId: String): Call<JobResponse>

    @PUT("api/jobs/{jobId}")
    fun updateJob(
        @Header("Authorization") token: String,
        @Path("jobId") jobId: String,
        @Body request: JobRequest
    ): Call<JobResponse>

    @DELETE("api/jobs/{jobId}")
    fun deleteJob(@Header("Authorization") token: String, @Path("jobId") jobId: String): Call<Void>

    @GET("api/matching/{jobId}")
    fun getMatchedWorkers(
        @Header("Authorization") token: String,
        @Path("jobId") jobId: String
    ): Call<List<MatchedWorker>>

    @POST("api/matching/search") // Changed to POST
    fun searchCandidates(
        @Header("Authorization") token: String,
        @Body request: CandidateSearchRequest
    ): Call<List<CandidateSearchResult>> // Changed to @Body

    // Applications
    @GET("api/applications/job/{jobId}")
    fun getApplicationsForJob(
        @Header("Authorization") token: String, @Path("jobId") jobId: String
    ): Call<List<JobApplicationItem>>

    @PUT("api/applications/{applicationId}/status")
    fun updateApplicationStatus(
        @Header("Authorization") token: String,
        @Path("applicationId") applicationId: String,
        @Query("status") status: String
    ): Call<JobApplicationItem>

    @POST("api/applications/offer")
    fun offerJob(
        @Header("Authorization") token: String,
        @Query("jobId") jobId: String,
        @Query("workerId") workerId: String
    ): Call<JobApplicationItem>

    // Notifications
    @GET("api/notifications")
    fun getUnreadNotifications(@Header("Authorization") token: String): Call<List<Notification>>

    @PUT("api/notifications/{notificationId}/read")
    fun markAsRead(@Header("Authorization") token: String, @Path("notificationId") notificationId: String): Call<Void>

    // Verification
    @GET("api/verification/pending")
    fun getPendingVerifications(@Header("Authorization") token: String): Call<List<Verification>>

    @GET("api/verification/worker/{workerId}")
    fun getWorkerDocuments(@Header("Authorization") token: String, @Path("workerId") workerId: String): Call<List<Verification>>

    // The actual uploaded files (filenames + view URLs) for a worker - what
    // an employer's profile view needs to show/open the real documents,
    // as opposed to getWorkerDocuments above, which despite its name only
    // returns coarse per-category verification status.
    @GET("api/verification/worker/{workerId}/documents")
    fun getWorkerDocumentFiles(
        @Header("Authorization") token: String,
        @Path("workerId") workerId: String
    ): Call<List<Document>>

    @PUT("api/verification/{verificationId}/status")
    fun updateEmployerVerificationStatus(
        @Header("Authorization") token: String,
        @Path("verificationId") verificationId: String,
        @Query("status") status: String,
        @Body comments: String
    ): Call<Void>

    // Skills
    @GET("api/skills")
    fun getSkills(): Call<List<Skill>>

    @GET("api/worker/profile/{workerId}")
    fun getWorkerProfile(@Header("Authorization") token: String, @Path("workerId") workerId: String): Call<WorkerProfileResponse>
}